test/unit/spec/EnrollWebauthn_spec.js

```diff
@@ -0,0 +1,304 @@
+        });
+      });
+
+      itp('shows a waiting spinner and devices images after submitting the form', function () {
+        spyOn(webauthn, 'isWebauthnOrU2fAvailable').and.returnValue(true);
+        mockWebauthnSuccessRegistration();
+        return setup().then(function (test) {
+          test.setNextResponse([resEnrollActivateWebauthn, resSuccess]);
+          test.form.submit();
+          return Expect.waitForSpyCall(test.successSpy, test);
+        })
+        .then(function (test) {
+          Expect.isVisible(test.form.enrollWaitingText());
+          Expect.isVisible(test.form.enrollDeviceImages());
+          Expect.isVisible(test.form.enrollSpinningIcon());
+          Expect.isNotVisible(test.form.submitButton());
+        });
+      });
+
+      itp('sends enroll request after submitting the form', function () {
+        spyOn(webauthn, 'isWebauthnOrU2fAvailable').and.returnValue(true);
+        mockWebauthnSuccessRegistration();
+        return setup().then(function (test) {
+          $.ajax.calls.reset();
+          test.setNextResponse([resEnrollActivateWebauthn, resSuccess]);
+          test.form.submit();
+          return Expect.waitForSpyCall(test.successSpy);
+        })
+        .then(function () {
+          expect($.ajax.calls.count()).toBe(2);
+          Expect.isJsonPost($.ajax.calls.argsFor(0), {
+            url: 'https://foo.com/api/v1/authn/factors',
+            data: {
+              stateToken: 'testStateToken',
+              factorType: 'webauthn',
+              provider: 'FIDO'
+            }
+          });
+        });
+      });
+
+      itp('calls navigator.credentials.create and activates the factor', function () {
+        spyOn(webauthn, 'isWebauthnOrU2fAvailable').and.returnValue(true);
+        spyOn(webauthn, 'isNewApiAvailable').and.returnValue(true);
+
+        mockWebauthnSuccessRegistration();
+        return setup().then(function (test) {
+          $.ajax.calls.reset();
+          test.setNextResponse([resEnrollActivateWebauthn, resSuccess]);
+          test.form.submit();
+          return Expect.waitForSpyCall(test.successSpy);
+        })
+        .then(function () {
+          expect(navigator.credentials.create).toHaveBeenCalledWith({
+            publicKey: {
+              rp: {
+                name: 'acme'
+              },
+              user: {
+                id: CryptoUtil.strToBin('00u1212qZXXap6Cts0g4'),
+                name: 'yuming.cao@okta.com',
+                displayName: 'Test User'
+              },
+              pubKeyCredParams: [{
+                type: 'public-key',
+                alg: -7
+              }],
+              challenge: CryptoUtil.strToBin('G7bIvwrJJ33WCEp6GGSH'),
+              authenticatorSelection: {
+                authenticatorAttachment: 'cross-platform',
+                requireResidentKey: false,
+                userVerification: 'preferred'
+              },
+              u2fParams: {
```

Comment:
do we need this in the data for `navigator.credentials.create`? If not, we should remove it

Reply:
See my comment https://github.com/okta/okta-signin-widget/pull/523#discussion_r227161299 for it's usage.